### PR TITLE
teodor-RM35074-add-PWA

### PR DESCRIPTION
**Pull request by Theodorson**

### CODE REVIEW

File: assets/javascripts/redmine_advanced_messenger.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const PWA_START_URL = "/my/page";
+const PWA_BADGE_VALUE = "PWA_BADGE_VALUE";
+const TIMER_CLOSING_OLD_NOTIFICATIONS = 5; // seconds
+
+// register service worker
+function registerPWAServiceWorker() {
+    if (navigator.serviceWorker) {
```
Comment (mircea-negreanu) [inline assets/javascripts/redmine_advanced_messenger.js:7]: As I see it the `serviceWorker` is register for normal use as well (not only on `PWA`).

Is it possible to register the `serviceWorker` only on PWA? Or is a request to work on Non PWA as well?

File: assets/javascripts/redmine_advanced_messenger.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        window.addEventListener("load", async function () {
+            try {
+                navigator.serviceWorker.register("../../plugin_assets/redmine_advanced_messenger/pwa/service-worker.js")
+                    .then((registration) => {
+                        console.log("Service worker registration succeeded:", registration);
```
Comment (mircea-negreanu) [inline assets/javascripts/redmine_advanced_messenger.js:12]: I'm curious where is shown this message ...

File: app/views/advanced_messenger/_html_head.html.erb [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
 <% content_for :header_tags do %>
     <%= stylesheet_link_tag 'advanced_messenger', :plugin => 'redmine_advanced_messenger' %>
+    <%# ============= PWA Manifest   ============= %>
+    <%= auto_discovery_link_tag :json, '../../plugin_assets/redmine_advanced_messenger/pwa/manifest.json', { rel: "manifest" } %>
 <% end %>  
+<%# ============= Plugin JS script  ============= %>
+<%= javascript_include_tag 'redmine_advanced_messenger', plugin: 'redmine_advanced_messenger' %>
+<%# ============= Register PWA service worker  ============= %>
+<%= javascript_tag do%>
+    registerPWAServiceWorker();
```
Comment (mircea-negreanu) [inline app/views/advanced_messenger/_html_head.html.erb:11]: I'm wondering if you can put here a check for whether or not you are running in a PWA and call the `registerPWAServiceWorker` only on PWA.

File: assets/javascripts/redmine_advanced_messenger.js (hunk 2 above)
Comment (Theodorson) [reply assets/javascripts/redmine_advanced_messenger.js:12]: I don't understand.

File: assets/javascripts/redmine_advanced_messenger.js (hunk 1 above)
Comment (Theodorson) [reply assets/javascripts/redmine_advanced_messenger.js:7]: It is possible to register just for PWA. OK from me to do this change.

File: app/views/advanced_messenger/_html_head.html.erb (hunk 3 above)
Comment (Theodorson) [reply app/views/advanced_messenger/_html_head.html.erb:11]: Same thing as above.

**Pull request merged**